pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -271,6 +285,15 @@ func (d *DeploymentController) Reconcile(req types.NamespacedName) error {
 		if !knownControllers.Contains(string(gc.Spec.ControllerName)) {
 			return nil
 		}
+		// find the tag or revision indicated by the object
+		// TODO: handle default tag
+		selectedTag, ok := gw.Labels[label.IoIstioRev.Name]
+		if !ok {
+			selectedTag = d.namespaces.Get(gw.Namespace, "").Labels[label.IoIstioRev.Name]
+		}
+		if !d.myTags.Contains(selectedTag) {
+			return nil
```

Comment:
we will need to make the leader election smarter as well I think, right now only 1 pod will even run this controller at a time

Reply:
Given https://github.com/istio/istio/pull/44174 this should actually be remove leader election I think.

Reply:
Will be a followup PR